tools/smoother_benchmarking/README.md

```diff
@@ -0,0 +1,121 @@
++      auto planning_duration = steady_clock_.now() - planning_start;
++      result->planning_time = planning_duration;
+ 
+       if (!validatePath<ActionThroughPoses>(curr_goal, curr_path, goal->planner_id)) {
+         throw nav2_core::NoValidPathCouldBeFound(goal->planner_id + "generated a empty path");
+@@ -398,7 +401,7 @@ void PlannerServer::computePlanThroughPoses()
+     publishPlan(result->path);
+ 
+     auto cycle_duration = steady_clock_.now() - start_time;
+-    result->planning_time = cycle_duration;
++    // result->planning_time = cycle_duration;
+ 
+     if (max_planner_duration_ && cycle_duration.seconds() > max_planner_duration_) {
+       RCLCPP_WARN(
+diff --git a/nav2_smoother/src/nav2_smoother.cpp b/nav2_smoother/src/nav2_smoother.cpp
+index ada1f664..610e9512 100644
+--- a/nav2_smoother/src/nav2_smoother.cpp
++++ b/nav2_smoother/src/nav2_smoother.cpp
+@@ -253,8 +253,6 @@ bool SmootherServer::findSmootherId(
+ 
+ void SmootherServer::smoothPlan()
+ {
+-  auto start_time = steady_clock_.now();
+-
+   RCLCPP_INFO(get_logger(), "Received a path to smooth.");
+ 
+   auto result = std::make_shared<Action::Result>();
+@@ -271,6 +269,8 @@ void SmootherServer::smoothPlan()
+     // Perform smoothing
+     auto goal = action_server_->get_current_goal();
+     result->path = goal->path;
++
++    auto start_time = steady_clock_.now();
+     result->was_completed = smoothers_[current_smoother_]->smooth(
+       result->path, goal->max_smoothing_duration);
+     result->smoothing_duration = steady_clock_.now() - start_time;
+```
+
+To use the suite, modify the Nav2 bringup parameters to include selected path planner:
+
+```
+planner_server:
+  ros__parameters:
+    expected_planner_frequency: 20.0
+    planner_plugins: ["SmacHybrid"]
+    SmacHybrid:
+      plugin: "nav2_smac_planner/SmacPlannerHybrid"
+      tolerance: 0.5
+      motion_model_for_search: "DUBIN" # default, non-reverse motion
+      smooth_path: false # should be disabled for experiment
+      analytic_expansion_max_length: 0.3 # decreased to avoid robot jerking
+```
+
+... and path smoothers for benchmark:
+
+```
+ smoother_server:
+   ros__parameters:
+    smoother_plugins: ["simple_smoother", "constrained_smoother"]
+    simple_smoother:
+      plugin: "nav2_smoother::SimpleSmoother"
+      holonomic: False # should be disabled to utilize motion/rotation model
+    constrained_smoother:
+      plugin: "nav2_constrained_smoother/ConstrainedSmoother"
+      w_smooth: 100000.0 # tuned
+```
+
+Then enable SmoothPath sequence for selected path planner, as shown for default `navigate_to_pose_w_replanning_and_recovery.xml` behavior tree:
+
+```
+       <PipelineSequence name="NavigateWithReplanning">
+         <RateController hz="1.0">
+           <RecoveryNode number_of_retries="1" name="ComputePathToPose">
+-            <ComputePathToPose goal="{goal}" path="{path}" planner_id="GridBased"/>
```

Comment:
Why modify the BT? You're just calling the task servers to get the path / smoothing results in the data collection script

Reply:
Indeed, this is only required when we are using standard operation flow. When calling actions from palth planner / smoother servers, we do not need it at all. Removed from README.